pandas/core/arrays/base.py

```diff
@@ -667,7 +672,12 @@ def argmax(self, skipna: bool = True) -> int:
             raise NotImplementedError
         return nargminmax(self, "argmax")
 
-    def fillna(self, value=None, method=None, limit=None):
+    def fillna(
+        self,
+        value: Optional[Union[Any, ArrayLike]] = None,
+        method: Optional[Literal["backfill", "bfill", "ffill", "pad"]] = None,
```

Comment:
I've not followed everything in the typing discussions so sorry if this has been discussed already, but is the idea that these will one day be available to users (e.g. by including a `py.typed` file)?

If so, then I wonder whether this might be too restrictive - e.g. if someone has `method: str` and they tried passing it here, they'd get an error like
```
Argument 1 to "foo" has incompatible type "str"; expected "Literal['backfill', 'bfill', 'ffill', 'pad']"  [arg-type]
```

If users are expected to use these literals, then perhaps `Literal["backfill", "bfill", "ffill", "pad"]` should be an alias (which users could import)?

Reply:
Yes, the idea is that these would be available to users.

But for something like `fillna`, I would assume that most people wouldn't pass a variable `method` as the argument, i.e., you're more likely to write `df.fillna(method = "ffill")` than 
```
def myfunc(df: pd.DataFrame, mymethod:str):
    df.fillna(method=mymethod)
```

Reply:
True - besides, this error message is perfectly clear so this is prob OK :+1: